`Couldn't create executionContexts` when joining system tables and filtering by `OBJECT` property

### CrateDB version

5.9.5 / today's nightly

### CrateDB setup information

Single node Docker container

### Problem description

There is an exception when trying to join two system tables and filtering by an attribute from an `OBJECT` column.

### Steps to Reproduce

```sql
CREATE TABLE t1 (
  ts TIMESTAMP,
  ts_month TIMESTAMP GENERATED ALWAYS AS DATE_TRUNC('month', ts),
  value INTEGER
)
PARTITIONED BY (ts_month);

SELECT values['ts_month']::TIMESTAMP, SUM(size)
FROM information_schema.table_partitions p
JOIN sys.shards s USING (partition_ident, table_name)
WHERE p.table_name = 't1'
  AND values['ts_month']::TIMESTAMP BETWEEN '2022-08-01'::TIMESTAMP AND '2024-01-01'::TIMESTAMP
GROUP BY 1
LIMIT 100;
```

There is no exception when eliminating the join or the `WHERE` condition on `values['ts_month']`.

### Actual Result

```
io.crate.exceptions.SQLParseException: Couldn't create executionContexts from
NodeOperations: [0=>NodeOp{ phase{id=0/collect, nodes=[F_hSIInuSJ6KBP3I0Rvh3Q], dist=MODULO}, downstreamNodes=[F_hSIInuSJ6KBP3I0Rvh3Q], downstreamPhase=4, downstreamInputId=0}]
Leafs: [JoinPhase{executionNodes=[F_hSIInuSJ6KBP3I0Rvh3Q], executionPhaseId=4, name=hash-join, joinType=INNER, joinCondition=((INPUT(1) = INPUT(4)) AND (INPUT(2) = INPUT(5))), outputTypes=[timestamp without time zone, bigint], jobId=280ef4b5-12d7-9453-264f-00393d938793, executionNodes=[F_hSIInuSJ6KBP3I0Rvh3Q]}]
target-sources: [4=>[0]]
original-error: UnsupportedOperationException: Can't handle Symbol [ScopedSymbol: values['ts_month']]
	at io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
	at io.crate.exceptions.SQLExceptions.prepareForClientTransmission(SQLExceptions.java:200)
	at io.crate.rest.action.SqlHttpHandler.sendResponse(SqlHttpHandler.java:176)
	at io.crate.rest.action.SqlHttpHandler.lambda$channelRead0$0(SqlHttpHandler.java:130)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:907)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:931)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2401)
	at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:128)
	at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:87)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.crate.protocols.http.HttpBlobHandler.channelRead0(HttpBlobHandler.java:159)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.crate.auth.HttpAuthUpstreamHandler.handleHttpChunk(HttpAuthUpstreamHandler.java:158)
	at io.crate.auth.HttpAuthUpstreamHandler.channelRead0(HttpAuthUpstreamHandler.java:95)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at org.elasticsearch.transport.netty4.Netty4InboundStatsHandler.channelRead(Netty4InboundStatsHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.lang.IllegalArgumentException: Couldn't create executionContexts from
NodeOperations: [0=>NodeOp{ phase{id=0/collect, nodes=[F_hSIInuSJ6KBP3I0Rvh3Q], dist=MODULO}, downstreamNodes=[F_hSIInuSJ6KBP3I0Rvh3Q], downstreamPhase=4, downstreamInputId=0}]
Leafs: [JoinPhase{executionNodes=[F_hSIInuSJ6KBP3I0Rvh3Q], executionPhaseId=4, name=hash-join, joinType=INNER, joinCondition=((INPUT(1) = INPUT(4)) AND (INPUT(2) = INPUT(5))), outputTypes=[timestamp without time zone, bigint], jobId=280ef4b5-12d7-9453-264f-00393d938793, executionNodes=[F_hSIInuSJ6KBP3I0Rvh3Q]}]
target-sources: [4=>[0]]
original-error: UnsupportedOperationException: Can't handle Symbol [ScopedSymbol: values['ts_month']]
	at io.crate.expression.BaseImplementationSymbolVisitor.visitSymbol(BaseImplementationSymbolVisitor.java:103)
	at io.crate.expression.BaseImplementationSymbolVisitor.visitSymbol(BaseImplementationSymbolVisitor.java:43)
	at io.crate.expression.symbol.SymbolVisitor.visitField(SymbolVisitor.java:66)
	at io.crate.expression.symbol.ScopedSymbol.accept(ScopedSymbol.java:116)
	at io.crate.expression.BaseImplementationSymbolVisitor.visitFunction(BaseImplementationSymbolVisitor.java:65)
	at io.crate.expression.BaseImplementationSymbolVisitor.visitFunction(BaseImplementationSymbolVisitor.java:43)
	at io.crate.expression.symbol.Function.accept(Function.java:251)
	at io.crate.expression.InputFactory$Context.add(InputFactory.java:159)
	at io.crate.expression.InputFactory$Context.add(InputFactory.java:144)
	at io.crate.execution.engine.pipeline.ProjectionToProjectorVisitor.visitGroupProjection(ProjectionToProjectorVisitor.java:330)
	at io.crate.execution.engine.pipeline.ProjectionToProjectorVisitor.visitGroupProjection(ProjectionToProjectorVisitor.java:144)
	at io.crate.execution.dsl.projection.GroupProjection.accept(GroupProjection.java:93)
	at io.crate.execution.dsl.projection.ProjectionVisitor.process(ProjectionVisitor.java:30)
	at io.crate.execution.engine.pipeline.ProjectionToProjectorVisitor.create(ProjectionToProjectorVisitor.java:748)
	at io.crate.execution.engine.pipeline.Projectors.<init>(Projectors.java:59)
	at io.crate.execution.engine.pipeline.ProjectingRowConsumer.<init>(ProjectingRowConsumer.java:101)
	at io.crate.execution.engine.pipeline.ProjectingRowConsumer.create(ProjectingRowConsumer.java:90)
	at io.crate.execution.jobs.JobSetup$InnerPreparer.visitHashJoinPhase(JobSetup.java:927)
	at io.crate.execution.jobs.JobSetup$InnerPreparer.visitHashJoinPhase(JobSetup.java:563)
	at io.crate.execution.dsl.phases.HashJoinPhase.accept(HashJoinPhase.java:120)
	at io.crate.execution.jobs.JobSetup.createContexts(JobSetup.java:258)
	at io.crate.execution.jobs.JobSetup.prepareOnHandler(JobSetup.java:245)
	at io.crate.execution.engine.JobLauncher.setupTasks(JobLauncher.java:240)
	at io.crate.execution.engine.JobLauncher.execute(JobLauncher.java:175)
	at io.crate.planner.operators.LogicalPlanner.executeNodeOpTree(LogicalPlanner.java:666)
	at io.crate.planner.operators.LogicalPlanner.doExecute(LogicalPlanner.java:604)
	at io.crate.planner.operators.LogicalPlanner.execute(LogicalPlanner.java:569)
	at io.crate.planner.operators.LogicalPlan.executeOrFail(LogicalPlan.java:216)
	at io.crate.planner.Plan.execute(Plan.java:74)
	at io.crate.session.Session.singleExec(Session.java:832)
	at io.crate.session.Session.execute(Session.java:583)
	at io.crate.rest.action.SqlHttpHandler.executeSimpleRequest(SqlHttpHandler.java:273)
	at io.crate.rest.action.SqlHttpHandler.handleSQLRequest(SqlHttpHandler.java:220)
	at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:127)
	... 50 more
Caused by: java.lang.UnsupportedOperationException: Can't handle Symbol [ScopedSymbol: values['ts_month']]
	... 84 more
```

### Expected Result

Successful execution, empty result set

--- a/docs/appendices/release-notes/5.9.6.rst
+++ b/docs/appendices/release-notes/5.9.6.rst
@@ -61,3 +61,14 @@ Fixes
   accuracy of the result can be now controlled and adjusted to the user's needs.
   Be aware that higher ``compression`` values will lead to slightly higher memory
   consumption.
+
+- Fixed a regression introduced with :ref:`version_5.8.5` which caused queries
+  containing a symbol pointing to an ignored column or dynamic system column of
+  an aliased relation to fail with an planner error. This happens only if the
+  related symbol was used multiple times, e.g. as a select item and inside the
+  where clause. Example::
+
+    SELECT values['ts_month']
+    FROM information_schema.table_partitions alias
+    WHERE values['ts_month'] = '2022-08-01'
+

--- a/server/src/main/java/io/crate/expression/symbol/ScopedSymbol.java
+++ b/server/src/main/java/io/crate/expression/symbol/ScopedSymbol.java
@@ -82,6 +82,14 @@ public final class ScopedSymbol implements Symbol {
     }
 
     /**
+     * Creates a ScopedSymbol describing the output from a relation, pointing to a single specific symbol of a child
+     * relation.
+     * <p>
+     * It uses the hashCode of the symbol to disambiguate columns with the same name but different source relations.
+     * Using the identity of the symbol is not possible because dynamic or void references are not cached and are
+     * created on the fly on every look up. This would lead to different identities for the same symbol.
+     * </p>
+     *
      * @param relation owner of this symbol
      * @param column name of the symbol
      * @param symbol the symbol this ScopedSymbol is pointing towards
@@ -90,7 +98,7 @@ public final class ScopedSymbol implements Symbol {
         this.relation = relation;
         this.column = column;
         this.dataType = symbol.valueType();
-        this.identity = System.identityHashCode(symbol);
+        this.identity = symbol.hashCode();
     }
 
     public RelationName relation() {

--- a/server/src/test/java/io/crate/planner/AliasPlannerTest.java
+++ b/server/src/test/java/io/crate/planner/AliasPlannerTest.java
@@ -1,0 +1,67 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner;
+
+import static io.crate.testing.Asserts.assertThat;
+
+import org.junit.Test;
+
+import io.crate.planner.node.dql.Collect;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+
+public class AliasPlannerTest extends CrateDummyClusterServiceUnitTest {
+
+    /**
+     * Tests a regression introduced by <a href="https://github.com/crate/crate/commit/1433bbaff379e1d20d4f4c54446d4c35b86cb63e">1433bba</a>
+     * causing to throw an exception.
+     * Relates to <a href="https://github.com/crate/crate/issues/17153">#17153</a>.
+     */
+    @Test
+    public void test_scoped_symbol_pointing_to_dynamic_system_sub_column_used_multiple_times() throws Exception {
+        var e = SQLExecutor.builder(clusterService)
+            .build();
+        Collect plan = e.plan("""
+            SELECT values['ts_month']
+            FROM information_schema.table_partitions alias
+            WHERE values['ts_month'] = '2022-08-01'
+            """);
+        assertThat(plan.collectPhase().toCollect().getFirst()).isDynamicReference();
+    }
+
+    /**
+     * Tests a regression introduced by <a href="https://github.com/crate/crate/commit/1433bbaff379e1d20d4f4c54446d4c35b86cb63e">1433bba</a>
+     * causing to throw an exception.
+     * Relates to <a href="https://github.com/crate/crate/issues/17153">#17153</a>.
+     */
+    @Test
+    public void test_scoped_symbol_pointing_to_ignored_sub_column_used_multiple_times() throws Exception {
+        var e = SQLExecutor.builder(clusterService).build()
+            .addTable("CREATE TABLE t (values OBJECT(IGNORED))");
+        Collect plan = e.plan("""
+            SELECT values['ts_month']
+            FROM t AS alias
+            WHERE values['ts_month'] = '2022-08-01'
+            """);
+        assertThat(plan.collectPhase().toCollect().getFirst()).isDynamicReference();
+    }
+}

